mmda/types/document.py

```diff
@@ -12,96 +12,140 @@
 from glob import glob
 
 from mmda.types.image import Image
-from mmda.types.document_elements import DocumentSymbols
-from mmda.types.annotation import Annotation, DocSpanGroup, Indexer, DocSpanGroupIndexer
+from mmda.types.annotation import Annotation, SpanGroup, Indexer, SpanGroupIndexer
+from mmda.types.names import Symbols, Images
 
 
-@dataclass
 class Document:
 
-    DEFAULT_FIELDS = ["symbols", "images"]
+    DEFAULT_FIELDS = [Symbols, Images]
+    UNALLOWED_FIELD_NAMES = ['fields']
 
     def __init__(
         self,
-        symbols: DocumentSymbols,
+        symbols: str,
         images: Optional[List["Image.Image"]] = None,
     ):
-
         self.symbols = symbols
         self.images = images
         self._fields = self.DEFAULT_FIELDS
         self._indexers: Dict[str, Indexer] = {}
 
     @property
-    def fields(self):
+    def fields(self) -> List[str]:
         return self._fields
 
-    def _check_valid_field_name(self, field_name):
-        assert not field_name.startswith(
-            "_"
-        ), "The field_name should not start with `_`. "
-        assert field_name not in ["fields"], "The field_name should not be 'fields'."
-        assert field_name not in dir(
-            self
-        ), f"The field_name should not conflict with existing class properties {field_name}"
-
-    def _register_field(self, field_name):
-        if field_name not in self.fields:
-            self._check_valid_field_name(field_name)
-            self._fields.append(field_name)
-            self._indexers[field_name] = DocSpanGroupIndexer(num_pages=self.symbols.page_count)
-
-    def _annotate(self, field_name, field_annotations):
-
-        self._register_field(field_name)
-        
-        for annotation in field_annotations:
-            annotation = annotation.annotate(self)
-            
-        setattr(self, field_name, field_annotations)
-
-    def annotate(self, **annotations: List[Annotation]):
+    # TODO: extend implementation to support DocBoxGroup
+    def find_overlapping(self, query: Annotation, field_name: str) -> List[Annotation]:
```

Comment:
why `find_overlapping` rather than `find`? because probably `find` will be used for other purposes?

Reply:
just for transparency sake. that this is an `overlapping` cal.  the reason for this is cuz of some discussion we had where how `find()` operates really is specific to this overlapping nature